src/spdx/parser/rdf/package_parser.py

```diff
@@ -0,0 +1,144 @@
+                                                    external_references=external_package_refs,
+                                                    attribution_texts=attribution_texts,
+                                                    primary_package_purpose=primary_package_purpose,
+                                                    release_date=release_date, built_date=built_date,
+                                                    valid_until_date=valid_until_date))
+
+    return package
+
+
+def parse_package_verification_code(package_verification_code_node: URIRef, graph: Graph) -> Optional[
+    PackageVerificationCode]:
+    logger = Logger()
+    value = parse_literal(logger, graph, package_verification_code_node, SPDX_NAMESPACE.packageVerificationCodeValue)
+    excluded_files = []
+    for (_, _, excluded_file_literal) in graph.triples(
+        (package_verification_code_node, SPDX_NAMESPACE.packageVerificationCodeExcludedFile, None)):
+        excluded_files.append(excluded_file_literal.toPython())
+
+    raise_parsing_error_if_logger_has_messages(logger)
+    package_verification_code = construct_or_raise_parsing_error(PackageVerificationCode, dict(value=value,
+                                                                                               excluded_files=excluded_files))
+    return package_verification_code
+
+
+def parse_external_package_ref(external_package_ref_node: URIRef, graph: Graph, doc_namespace) -> ExternalPackageRef:
+    logger = Logger()
+    ref_locator = parse_literal(logger, graph, external_package_ref_node, SPDX_NAMESPACE.referenceLocator)
+    ref_category = parse_literal(
+        logger, graph, external_package_ref_node, SPDX_NAMESPACE.referenceCategory,
+        parsing_method=lambda x: parse_enum_value(x, ExternalPackageRefCategory, SPDX_NAMESPACE.referenceCategory_, ))
+    ref_type = parse_literal(logger, graph, external_package_ref_node, SPDX_NAMESPACE.referenceType,
+                             parsing_method=lambda x: parse_external_package_ref_type(x, doc_namespace))
+    comment = parse_literal(logger, graph, external_package_ref_node, RDFS.comment)
+
+    raise_parsing_error_if_logger_has_messages(logger)
+    external_package_ref = construct_or_raise_parsing_error(ExternalPackageRef,
+                                                            dict(category=ref_category, reference_type=ref_type,
+                                                                 locator=ref_locator, comment=comment))
+    return external_package_ref
+
+def parse_external_package_ref_type(external_package_ref_type_literal: Literal, doc_namespace: str) -> str:
+    if external_package_ref_type_literal.startswith(doc_namespace):
+        return external_package_ref_type_literal[len(doc_namespace)+1:]
+    if external_package_ref_type_literal.startswith(REFERENCE_NAMESPACE):
+        return external_package_ref_type_literal[len(REFERENCE_NAMESPACE):]
```

Comment:
Would it be more consistent to use the `fragment` functions to extract the fragment here? But then, there might be some weird `external_package_ref_type_literal` that starts with the `doc_namespace` but has no `#` in it. Hmm...

Reply:
I think we can assume that the `doc_namespace` is separated with `#`. We can't use fragment for the `REFERENCE_NAMESPACE` as it is not separated with `#`.